pkg/driver/controller.go

```diff
@@ -438,7 +439,15 @@ func (d *Driver) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequ
 	opts := cloud.SnapshotOptions{
 		SnapshotName:   &snapshotName,
 		SourceVolumeId: &volumeID,
-		Tags:           &tags,
+	}
+
+	for key, value := range snapshotParams {
+		switch key {
+		case snapshotParamsTags:
+			opts.Tags = util.StringToStringPointer(value)
+		default:
+			return nil, status.Errorf(codes.InvalidArgument, "invalid parameter of %s: %s", key, value)
```

Comment:
We can't utilize this due to parameter reservations https://kubernetes-csi.github.io/docs/external-snapshotter.html

Reply:
Re-tested and confirmed that this implementation actually works. This is probably fine to merge after I address your nit.

Reply:
Sounds good. If for some reason a dependency update in the future breaks this functionality our E2E tests should pick it up. If we want to be proactive we can also perform a prefix check here, but I won't block on it.

Reply:
StringtoStringPointer is something that can be handled by the aws api: https://docs.aws.amazon.com/sdk-for-go/api/aws/ . Please fix this to you the aws call instead